app/models/service_template_terraform_template.rb

```diff
@@ -41,4 +58,39 @@ def terraform_template(action)
 
     ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Template.find(template_id)
   end
+
+  def create_dialogs(config_info)
+    dialog_hash = {}
+
+    info = config_info[:provision]
+    if info
+      # create new dialog, if required for :provision action
+      if info.key?(:new_dialog_name) && !info.key?(:dialog_id)
+        provision_dialog_id = create_new_dialog(info[:new_dialog_name], terraform_template(:provision), info[:extra_vars]).id
+        dialog_hash[:provision] = {:dialog_id => provision_dialog_id}
+      else
+        provision_dialog_id = info[:dialog_id]
+      end
+
+      # For :retirement & :reconfigure,  we use the same dialog as in :provision action
+      dialog_hash = [:retirement, :reconfigure].each_with_object(dialog_hash) do |action, hash|
+        hash[action] = {:dialog_id => provision_dialog_id}
+      end
+    end
+
+    dialog_hash
+  end
+
+  private
+
+  def create_new_dialog(dialog_name, terraform_template, extra_vars)
+    Dialog::TerraformTemplateServiceDialog.create_dialog(dialog_name, terraform_template, extra_vars)
+  end
+
+  def validate_update_config_info(options)
+    opts = super
+    return unless options.key?(:config_info)
+
+    self.class.send(:validate_config_info, opts)
```

Comment:
If you need to call this method from an instance variable I'd prefer you make it not a private_class_method and not use send